src/skalo/positioning.rs

```diff
@@ -0,0 +1,256 @@
+                }
+            }
+        }
+    }
+
+    // determine the most frequent position and its count for forward and reverse-complement
+    let forward = if !vec_position_forward.is_empty() {
+        let result = most_frequent_position(&vec_position_forward);
+        if result.1 == 0 {
+            None
+        } else {
+            Some(result)
+        }
+    } else {
+        None
+    };
+
+    let reverse = if !vec_position_reverse.is_empty() {
+        let result = most_frequent_position(&vec_position_reverse);
+        if result.1 == 0 {
+            None
+        } else {
+            Some(result)
+        }
+    } else {
+        None
+    };
+
+    // determine the final position and orientation
+    let (positioned, final_orientation) = match (forward, reverse) {
+        (Some((pos_forward, count_forward)), Some((pos_reverse, count_reverse))) => {
+            match count_forward.cmp(&count_reverse) {
+                std::cmp::Ordering::Equal => (false, "none".to_string()),
+                std::cmp::Ordering::Greater => {
+                    final_position = pos_forward;
+                    (true, "for".to_string())
+                }
+                std::cmp::Ordering::Less => {
+                    final_position = pos_reverse;
+                    (true, "rc".to_string())
+                }
+            }
+        }
+        (Some((pos_forward, _)), None) => {
+            final_position = pos_forward;
+            (true, "for".to_string())
+        }
+        (None, Some((pos_reverse, _))) => {
+            final_position = pos_reverse;
+            (true, "rc".to_string())
+        }
+        (None, None) => (false, "none".to_string()),
+    };
+
+    (positioned, final_position, final_orientation)
+}
+
+// returns the most frequent position and its count, or (0, 0) if there's a tie
+fn most_frequent_position(numbers: &[u32]) -> (u32, usize) {
+    let counts = numbers.iter().fold(HashMap::new(), |mut counts, &num| {
+        *counts.entry(num).or_insert(0) += 1;
+        counts
+    });
+
+    let mut max_element = None;
+    let mut max_count = 0;
+    let mut tie = false;
+
+    for (&num, &count) in &counts {
+        match count.cmp(&max_count) {
+            Ordering::Greater => {
+                max_element = Some(num);
+                max_count = count;
+                tie = false;
+            }
+            Ordering::Equal => {
+                tie = true;
+            }
+            Ordering::Less => {}
+        }
+    }
+
+    if tie {
+        return (0, 0); // return (0, 0) in case of a tie
+    }
+
+    // if no tie, return the most frequent position and its count
+    if let Some(position) = max_element {
+        if max_count < 10 {
+            return (0, 0); // return (0, 0) in case of low positioning
+        } else {
+            return (position, max_count);
+        }
+    }
+
+    (0, 0) // default return if input is empty
+}
```

Comment:
This logic can be tidied up I think, aim for a single `return` if possible

i.e. end with (position, max_count) and set them to zero when you have a tie or max_count is low